service/raft/group0_voter_handler.cc

```diff
@@ -481,7 +482,7 @@ future<> group0_voter_handler::update_nodes(
 
     group0_voter_calculator::nodes_list_t nodes;
 
-    auto get_replica_state = [this](const raft::server_id& id, bool allow_transitioning) -> const replica_state* {
+    auto get_replica_state = [this](const raft::server_id& id, raft::can_vote_t allow_transitioning) -> const replica_state* {
```

Comment:
gleb commented about this before, seems strange that parameter named `allow_transitioning` is typed `raft::can_vote_t `

Reply:
true, but it was expecting this type

Reply:
`allow_transitioning` just happened to be initialized from `group0_config.can_vote(id);` expression, it doesn't mean it's about `can_vote_t` is its essence. Better to just cast to bool as the initialization.

Reply:
so this is kinda funny , it initializes from the expression that asks if we can vote and uses can_vote_t but we want to cast it to bool here. but the bool is actually the bool class for can_vote, i dont mind changing it but you see the inconsistency righ?